### DISCUSSION TITLE
Synthetic dataset generation - Brief howto built on top of llama.cpp

### DISCUSSION THREAD

**Discussion by paschembri** [ggml-org/llama.cpp#3568] [Show and tell]
I’am currently exploring finetuning smaller models for classification / fill-mask.

I wrote a quick intro on how to produce synthetic dataset in [a few lines of python](https://advanced-stack.com/resources/how-to-generate-synthetic-dataset-using-a-llm.html).

It covers:
- the raw generation
- the automatic labelling/classification

If you have performance tips for this kind of tasks I’m all ears 👀. Now I have to look on how to implement distilbert using ggml 😱

**Comment by ianscrivener** [comment]
Suggest you first explore [**grammer functionality**](https://github.com/search?q=repo%3Aggerganov%2Fllama.cpp+grammar&type=discussions) in llama.cpp and the excellent [grammer builder](https://grammar.intrinsiclabs.ai/) helper app. I suspect you'll find them valuable.

**Reply by paschembri** [reply]
I did and it's great - I'm currently looking on how to backport the feature to the ggml lib (https://github.com/ggerganov/llama.cpp/discussions/2494)
